core/schema_utils.py

```diff
@@ -682,3 +682,28 @@ def is_valid_username_string(obj: str) -> bool:
             return True
         except utils.ValidationError:
             return False
+
+    @staticmethod
+    def should_end_with(
+        obj: str,
+        choices: List[str],
+        match_case: Optional[bool] = False
+    ) -> bool:
+        """Checks if the given obj (a string) is a filename with
+        valid audio extension.
+
+        Args:
+            obj: str. The string to verify.
+            choices: List[str]. All valid extensions.
+            match_case: bool. Whether the match is case sensitive.
+
+        Returns:
+            bool. Whether the given object is a filename with valid
+            audio extension.
+        """
+        dot_index = obj.rfind('.')
+        extension = obj[dot_index + 1:]
+        if match_case is False:
+            extension = extension.lower()
+            choices = [choice.lower() for choice in choices]
```

Comment:
Please use a general term for these variables as the function doesn't only apply to the audio extension.